src/features/orders/Timeline.tsx

```diff
@@ -0,0 +1,141 @@
+import DetailLongTextWrapper from './DetailLongTextWrapper'
+import Box from '@material-ui/core/Box'
+import { MAP_STATUS_ID_TO_TEXT } from './constants'
+import RejectedIcon from '@material-ui/icons/Close'
+import { types } from './reducer'
+
+const DISABLE_COLOR = '#9e9e9e';
+
+const useTimelineStyles = makeStyles((theme: Theme) => ({
+    contentWrapper: {
+        width: '35em',
+        [theme.breakpoints.down('xs')]: {
+            width: '15em',
+        },
+    },
+}));
+
+const useDoneStyles = makeStyles({
+    icon: {
+        color: '#0cb500',
+    },
+});
+
+export default function Timeline() {
+    const classes = useTimelineStyles();
+    const dispatch = useDispatch();
+    const theme = useTheme();
+
+    const activeStep = useSelector(selectors.timeline.activeStep);
+    const orderId = useSelector(selectors.detail.id);
+    const order = useSelector(selectors.ordersTable.data)[orderId];
+    const status = order.order_status;
+
+    const handleStepChange = useCallback(
+        (id: number) => {
+            dispatch(actions.timeline.changeActiveStep(id));
+        },
+        [dispatch],
+    );
+
+    useEffect(() => {
+        dispatch(
+            actions.timeline.changeActiveStep(statusToStep(order.order_status)),
+        );
+    }, [order, dispatch]);
+
+    return (
+        <TimelineComponent
+            steps={[
+                {
+                    label: MAP_STATUS_ID_TO_TEXT[1],
+                    icon: <LocalShippingIcon></LocalShippingIcon>,
+                    content: (
+                        <Box className={classes.contentWrapper}>
+                            <DetailLongTextWrapper title='Order date'>
+                                {order.order_date}
+                            </DetailLongTextWrapper>
+                        </Box>
+                    ),
+                    status: status === 1 ? <div></div> : <Done></Done>,
+                }, {
+                    label: MAP_STATUS_ID_TO_TEXT[2],
+                    icon: <SendIcon></SendIcon>,
+                    state: (status > 1) ? 'enable' : 'disable',
+                    content: (
+                        <Box className={classes.contentWrapper}>
+                            <DetailLongTextWrapper title='Required date'>
+                                {order.required_date}
+                            </DetailLongTextWrapper>
+                        </Box>
+                    ),
+                    status: (status > 2) ? <Done></Done> : <div></div>,
+                    color: (status > 1) ?
+                        theme.palette.primary.main : DISABLE_COLOR,
+                },
+                lastStepProps(order, theme, classes.contentWrapper),
+            ]}
+            activeStep={activeStep < 0 ? statusToStep(order.order_status) : activeStep}
+            onStepChange={handleStepChange}
+        ></TimelineComponent>
+    );
+}
+
+function statusToStep(status: number): number {
```

Comment:
lol, this function is funny) I didn't understand what it is :)

Reply:
at least it's better to define constants for statuses. now it looks like magic numbers